includes/wc-product-functions.php

```diff
@@ -751,3 +751,123 @@ function wc_get_product_visibility_options() {
 		'hidden'  => __( 'Hidden', 'woocommerce' ),
 	) );
 }
+
+/**
+ * Get relateed products.
+ * @param  integer  $product_id
+ * @param  integer  $limit
+ * @param  array    $exclude_ids
+ * @return array
+ */
+function wc_get_related_products( $product_id, $limit = 5, $exclude_ids = array() ) {
+	global $wpdb;
+
+	$product_id     = absint( $product_id );
+	$exclude_ids    = array_map( 'absint', array_merge( array( 0, $product_id ), $exclude_ids ) );
+	$transient_name = 'wc_related_' . $product_id;
+	$related_posts  = get_transient( $transient_name );
+	$limit          = $limit > 0 ? $limit : 5;
+
+	// We want to query related posts if they are not cached, or we don't have enough
+	if ( false === $related_posts || sizeof( $related_posts ) < $limit ) {
+		// Related products are found from category and tags
+		if ( apply_filters( 'woocommerce_product_related_posts_relate_by_category', true, $product_id ) ) {
+			$cats_array = wc_get_related_terms( $product_id, 'product_cat' );
+		} else {
+			$cats_array = array();
+		}
+
+		if ( apply_filters( 'woocommerce_product_related_posts_relate_by_tag', true, $product_id ) ) {
+			$tags_array = wc_get_related_terms( $product_id, 'product_tag' );
+		} else {
+			$tags_array = array();
+		}
+
+		// Don't bother if none are set, unless woocommerce_product_related_posts_force_display is set to true in which case all products are related.
+		if ( empty( $cats_array ) && empty( $tags_array ) && ! apply_filters( 'woocommerce_product_related_posts_force_display', false, $product_id ) ) {
+			$related_posts = array();
+		} else {
+			// Generate query - but query an extra 10 results to give the appearance of random results
+			$query = apply_filters( 'woocommerce_product_related_posts_query', wc_get_related_products_query( $cats_array, $tags_array, $exclude_ids, $limit + 10 ), $product_id );
+
+			// Get the posts
+			$related_posts = $wpdb->get_col( implode( ' ', $query ) );
+		}
+
+		set_transient( $transient_name, $related_posts, DAY_IN_SECONDS );
+	}
+
+	// Randomise the results
```

Comment:
We don't need some of the extra obvious comments here like "randomise the results", "limit the results", or "get the posts".